app/main.py

```diff
@@ -1,8 +1,34 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+    def add_attribute(self, attribute: str, value: object) -> None:
+        setattr(self, attribute, value)
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result_people_list = []
+
+    for person_dict in people:
+        Person.people[person_dict["name"]] = Person(person_dict["name"],
+                                                    person_dict["age"])
+
+    for person in people:
+        person_obj = Person.people.get(person["name"])
+
+        wife = person.get("wife")
+        if wife:
+            wife_obj = Person.people.get(wife)
+            person_obj.add_attribute("wife", wife_obj)
+
+        husband = person.get("husband")
+        if husband:
+            husband_obj = Person.people.get(husband)
+            person_obj.add_attribute("husband", husband_obj)
+
+        result_people_list.append(person_obj)
```

Comment:
```suggestion
    for person in people:
        person_obj = Person.people.get(person["name"])

        if person.get("wife"):
            wife_obj = Person.people.get(wife)
            person_obj.add_attribute("wife", wife_obj)

        if person.get("husband"):
            husband_obj = Person.people.get(husband)
            person_obj.add_attribute("husband", husband_obj)

        result_people_list.append(person_obj)
```
  You can simplify a bit